core/src/main/java/com/mapzen/android/graphics/MapInitializer.java

```diff
@@ -90,15 +89,13 @@ private void loadMap(final MapView mapView, String sceneFile, final OnMapReadyCa
     final List<SceneUpdate> sceneUpdates = sceneUpdateManager.getUpdatesFor(apiKey, locale,
         mapStateManager.isTransitOverlayEnabled(), mapStateManager.isBikeOverlayEnabled(),
         mapStateManager.isPathOverlayEnabled());
-    getTangramView(mapView).getMapAsync(new com.mapzen.tangram.MapView.OnMapReadyCallback() {
-      @Override public void onMapReady(MapController mapController) {
-        mapController.setHttpHandler(mapzenMapHttpHandler.httpHandler());
-        MapzenManager mapzenManager = MapzenManager.instance(mapView.getContext());
-        callback.onMapReady(
-            new MapzenMap(mapView, mapController, new OverlayManager(mapView, mapController,
-                mapDataManager, mapStateManager), mapStateManager, new LabelPickHandler(mapView),
-                new MarkerManager(mapController), sceneUpdateManager, locale, mapzenManager));
+    MapController controller = mapView.getTangramMapView().getMap(
+        new MapController.SceneLoadListener() {
+      @Override public void onSceneReady(int sceneId, SceneError sceneError) {
+        mapReadyInitializer.onMapReady(mapView, mapzenMapHttpHandler, callback, mapDataManager,
```

Comment:
Because of the way our APIs are setup I don't think this is needed.

The flow the user goes through is either:
1. `MapView#getMapAsync(OnMapReadyCallback)` or `MapFragment#getMapAsync(OnMapReadyCallback)`
2. `other calls to the map after receiving it in OnMapReadyCallback`

We wrap all calls to `MapController` so it is only after receiving this callback that the user has a `MapzenMap` and can make additional calls to `MapController`. This means that the first call we get to `onSceneReady` is guaranteed to be the one we care about because it is in response to the only call we have made to `MapController` (unless the user has accessed `MapController` directly and made calls to it. But this case can lead to breaking other parts of our API that we don't support).